### PR TITLE
Stop showing devtools reload message for Boot >2.4

### PR DESCRIPTION
**Pull request by pleku**
The issue has been fixed in Spring Boot 2.4.0.
Bumps the project version to Spring Boot 2.4.0.

### CODE REVIEW

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/router/SpringRouteNotFoundError.java [hunk 1]
```diff
@@ -5,24 +5,31 @@
...
-                    + "lost due to a <a href ='https://github.com/spring-projects/spring-boot/issues/19543'>"
-                    + "compilation race condition</a>. See "
-                    + "<a href='https://vaadin.com/docs/flow/workflow/setup-live-reload-springboot.html'>"
-                    + "the documentation</a> for further workarounds and other "
-                    + "live reload alternatives.";
-            getElement().appendChild(new Html(customMessage).getElement());
+            // This has been fixed since Spring Boot 2.4.0
+            String springBootVersion = SpringBootVersion.getVersion();
+            // the version is e.g. "2.2.0.RELEASE" or "2.4.0" ...
+            final String[] numbers = springBootVersion == null ? null : springBootVersion.split("\\.");
```
Comment (caalador) [inline vaadin-spring/src/main/java/com/vaadin/flow/spring/router/SpringRouteNotFoundError.java:22]:
```
if(springBootVersion == null) {
  return;
}
final FrontendVersion springVersion = new FrontendVersion(springBootVersion);
if (springVersion.isOlderThan(new FrontendVersion(2,4,0))) { ...
```

**Review by caalador — CHANGES_REQUESTED** [review]

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/router/SpringRouteNotFoundError.java (hunk 1 above)
Comment (pleku) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/router/SpringRouteNotFoundError.java]: Done

**Review by caalador — DISMISSED** [review]

**Comment by pleku** [comment]
@mshabarov I had to do some changes to the PowerMock usage since it is not compatible with newer Mockito versions, can you please check the test changes in VaadinServletContextInitializerTest.java

File: pom.xml [hunk 2]
```diff
@@ -45,15 +45,15 @@
 
         <!-- These are typically overridden with BOMs -->
         <vaadin.flow.version>6.0-SNAPSHOT</vaadin.flow.version>
-        <spring-boot.version>2.2.0.RELEASE</spring-boot.version>
+        <spring-boot.version>2.4.0.RELEASE</spring-boot.version>
```
Comment (caalador) [inline pom.xml:48]: Shouldn't it be now `2.4.0` not `2.4.0.RELEASE` ?

**Review by caalador — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (pleku) [reply pom.xml]: 🤦 yes

**Review by caalador — APPROVED** [review]

**Comment by pleku** [comment]
Cherry-picked to 15.0

**Pull request merged**